### PR TITLE
BFD-3241: Removes support for SKIPPED_RIF_RECORDS

### PR DESCRIPTION
**Pull request by meliGuzman**
**JIRA Ticket:**
[BFD-3241](https://jira.cms.gov/browse/BFD-3241)

**User Story or Bug Summary:**

Remove support for SKIPPED_RIF_RECORDS since there's no longer a need for tracking Beneficiary RIF records where REF_YR did not match the current processing year.

---

### What Does This PR Do?

- Removes ORM reference to SKIPPED_RIF_RECORDS in Beneficiary.yaml
- Removes references to SKIPPED_RIF_RECORDS in transformers and providers

Note: There does not exist a foreign key in the BENEFICIARIES table to the SKIPPED_RIF_RECORDS table.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/QueryLoggingListener.java [hunk 1]
```diff
@@ -110,6 +110,7 @@ public void beforeQuery(ExecutionInfo execInfo, List<QueryInfo> queryInfoList) {
     // Nothing to do here.
   }
 
+  // TODO: GET RID OF JOINS, got rid of left join, check if need join with bene hist table
```
Comment (aschey-forpeople) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/QueryLoggingListener.java:113]: Is this TODONE?

**Comment by aschey-forpeople** [comment]
Has this been tested in an ephemeral environment? Might be a good idea since we have some database changes.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/QueryLoggingListener.java (hunk 1 above)
Comment (meliGuzman) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/QueryLoggingListener.java]: Yes lol Removed!

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/PatientE2EBase.java [hunk 2]
```diff
@@ -821,7 +821,8 @@ public void testPatientByLogicalIdWithNoResultsAndPaginationRequestedExpect200()
    * expected for Patient by mbi.
    */
   @Test
-  public void testPatientByIdentifierHasAccessJsonWithExpectedMdcKeys() throws IOException {
+  public void testPatientByIdentifierHasAccessJsonWithExpectedMdcKeys()
+      throws IOException { // problematic one
```
Comment (aschey-forpeople) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/PatientE2EBase.java:825]: I think we can remove this comment too 
```suggestion
      throws IOException {
```

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/PatientE2EBase.java [hunk 3]
```diff
@@ -831,7 +832,7 @@ public void testPatientByIdentifierHasAccessJsonWithExpectedMdcKeys() throws IOE
             + "|"
             + mbiHash;
 
-    List<String> additionalExpectedMdcKeys = new ArrayList<>(MDC_EXPECTED_BASE_KEYS);
+    List<String> additionalExpectedMdcKeys = new ArrayList<>();
```
Comment (aschey-forpeople) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/PatientE2EBase.java:835]: Did we want to remove all of these? I thought it was just the IncludeIdentifiers one that was problematic.

Comment (aschey-forpeople) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/PatientE2EBase.java:834]: Oh my bad, I thought some of those keys weren't affected.

Comment (meliGuzman) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/PatientE2EBase.java:834]: @aschey-forpeople Just HTTP_ACCESS_RESPONSE_DURATION_PER_KB is unaffected in this test and HTTP_ACCESS_RESPONSE_HEADER_CONTENT_LOCATION in another

**Comment by aschey-forpeople** [comment]
Just noticed one more thing. There's a reference to the table in a query here: https://github.com/CMSgov/beneficiary-fhir-data/blob/master/apps/utils/db-query-checker/src/db_queries/select_bene_records_by_bene_ids.sql used by the db query checker. We may want to go ahead and remove that.

**Review by dondevun — DISMISSED** [review]

**Review by mjburling — APPROVED** [review]

**Review by aschey-forpeople — APPROVED** [review]

**Pull request merged**